mmv1/templates/terraform/custom_delete/monitoring_uptime_check_config.go.erb

```diff
@@ -0,0 +1,55 @@
+billingProject := ""
+
+project, err := tpgresource.GetProject(d, config)
+if err != nil {
+    return fmt.Errorf("Error fetching project for UptimeCheckConfig: %s", err)
+}
+billingProject = project
+
+lockName, err := tpgresource.ReplaceVars(d, config, "stackdriver/groups/{{project}}")
+if err != nil {
+    return err
+}
+transport_tpg.MutexStore.Lock(lockName)
+defer transport_tpg.MutexStore.Unlock(lockName)
+
+url, err := tpgresource.ReplaceVars(d, config, "{{MonitoringBasePath}}v3/{{name}}")
+if err != nil {
+    return err
+}
+
+var obj map[string]interface{}
+log.Printf("[DEBUG] Deleting UptimeCheckConfig %q", d.Id())
+
+// err == nil indicates that the billing_project value was found
+if bp, err := tpgresource.GetBillingProject(d, config); err == nil {
+    billingProject = bp
+}
+
+res, err := transport_tpg.SendRequest(transport_tpg.SendRequestOptions{
+    Config:               config,
+    Method:               "DELETE",
+    Project:              billingProject,
+    RawURL:               url,
+    UserAgent:            userAgent,
+    Body:                 obj,
+    Timeout:              d.Timeout(schema.TimeoutDelete),
+    ErrorRetryPredicates: []transport_tpg.RetryErrorPredicateFunc{transport_tpg.IsMonitoringConcurrentEditError},
+})
+
+
+<%#
+    The generated DELETE method returns a generic error for 400. 
+    Need to include a help message about deleting associated Alert Policies.
+-%>
+
+if err != nil {
+    if transport_tpg.IsGoogleApiErrorWithCode(err, 400) {
+        err := fmt.Errorf("%w Make sure all associated Alert Policies are deleted first.", err)
```

Comment:
My worry here is that we are checking a 400 and then saying `Make sure all associated Alert Policies are deleted first`, as though that is the cause of the 400. How do we know that this is the case? If we are not sure maybe we can make the statement less authoritative?

```suggestion
        err := fmt.Errorf("%w - please ensure all Alert Policies are deleted.", err)
```

Reply:
Done